samples/bot-sso-adaptivecard/csharp/BotSsoAdaptivecard/Bots/DialogBot.cs

```diff
@@ -0,0 +1,294 @@
+                var initialAdaptiveCard = GetFirstOptionsAdaptiveCard(path, signInLink, turnContext.Activity.From.Name, member.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(initialAdaptiveCard), cancellationToken);
+            }
+            else if (turnContext.Activity.Text.Contains("ABSSORefresh"))
+            {
+                string[] striPath = { ".", "Resources", "ABSSORefresh.json" };
+                var varMember = await TeamsInfo.GetMemberAsync(turnContext, turnContext.Activity.From.Id, cancellationToken);
+                var varInitialAdaptiveCard = GetFirstOptionsAdaptiveCard(striPath, signInLink, turnContext.Activity.From.Name, varMember.Id);
+                await turnContext.SendActivityAsync(MessageFactory.Attachment(varInitialAdaptiveCard), cancellationToken);
+            }
+            else
+            {
+                await turnContext.SendActivityAsync(MessageFactory.Text("Please send 'login' for options"), cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// The OAuth Prompt needs to see the Invoke Activity in order to complete the login process.
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected override async Task<InvokeResponse> OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (turnContext.Activity.Name == "signin/verifyState")
+            {
+                _logger.LogInformation("Running dialog with signin/verifystate from an Invoke Activity.");
+
+                // Run the Dialog with the new Invoke Activity.
+                await _dialog.RunAsync(turnContext, _conversationState.CreateProperty<DialogState>(nameof(DialogState)), cancellationToken);
+            }
+
+            else if (turnContext.Activity.Name == "adaptiveCard/action")
+            {
+
+                if (turnContext.Activity.Value == null)
+                    return null;
+
+                JObject value = JsonConvert.DeserializeObject<JObject>(turnContext.Activity.Value.ToString());
+
+                if (value["action"] == null)
+                    return null;
+
+                JObject actiondata = JsonConvert.DeserializeObject<JObject>(value["action"].ToString());
+
+                if (actiondata["verb"] == null)
+                    return null;
+
+                string verb = actiondata["verb"].ToString();
+                JObject authentication = null;
+                if (value["authentication"] != null)
+                {
+                    authentication = JsonConvert.DeserializeObject<JObject>(value["authentication"].ToString());
+                    string token = authentication["token"].ToString();
+                    var userTokenClient = turnContext.TurnState.Get<UserTokenClient>();
```

Comment:
remove line 140, 141. Just print the "token" to console.

Reply:
I have updated your comment. Could you please review and update.